homeassistant/components/whirlpool/sensor.py

```diff
@@ -237,3 +337,105 @@ def native_value(self) -> datetime | None:
             ):
                 self._value = new_timestamp
         return self._value
+
+
+class WhirlpoolOvenCavitySensor(WhirlpoolOvenEntity, SensorEntity):
+    """A class for Whirlpool oven cavity sensors."""
+
+    def __init__(
+        self,
+        oven: Oven,
+        cavity: Cavity,
+        description: WhirlpoolOvenCavitySensorEntityDescription,
+    ) -> None:
+        """Initialize the oven cavity sensor."""
+        super().__init__(oven)
+        cavity_key_suffix = self.get_cavity_key_suffix(cavity)
+        self.cavity = cavity
+        self.entity_description: WhirlpoolOvenCavitySensorEntityDescription = (
+            description
+        )
+        self._attr_unique_id = f"{oven.said}_{description.key}{cavity_key_suffix}"
+        self._attr_translation_key = f"{description.key}{cavity_key_suffix}"
+        self._attr_device_class = getattr(description, "device_class", None)
+        self._attr_state_class = getattr(description, "state_class", None)
+        self._attr_native_unit_of_measurement = getattr(
+            description, "native_unit_of_measurement", None
+        )
+        self._attr_options = getattr(description, "options", None)
+
+    @property
+    def native_value(self) -> StateType | str | int | float | None:
+        """Return native value of sensor."""
+        return self.entity_description.value_fn(self.oven, self.cavity)
+
+
+class WhirlpoolOvenCavityTimeSensor(WhirlpoolOvenEntity, RestoreSensor):
+    """A timestamp class for a Whirlpool oven cavity."""
+
+    _attr_should_poll = True
+
+    def __init__(
+        self,
+        oven: Oven,
+        cavity: Cavity,
+        description: WhirlpoolOvenCavitySensorEntityDescription,
+    ) -> None:
+        """Initialize the oven cavity time sensor."""
+        super().__init__(oven)
+        cavity_key_suffix = self.get_cavity_key_suffix(cavity)
+        self.cavity = cavity
+        self.entity_description: WhirlpoolOvenCavitySensorEntityDescription = (
+            description
+        )
+        self._attr_unique_id = f"{oven.said}_{description.key}{cavity_key_suffix}"
+        self._attr_translation_key = f"{description.key}{cavity_key_suffix}"
+        self.cook_time_duration: int = 0
+        self.cook_time_elapsed: int = 0
+        self.value: datetime | None = None
+
+    async def async_added_to_hass(self) -> None:
+        """Register attribute updates callback."""
+        restored_data = await self.async_get_last_sensor_data()
+        if restored_data and isinstance(restored_data.native_value, datetime):
+            self.value = restored_data.native_value
+        self.oven.register_attr_callback(self.on_attr_change)
+        await super().async_added_to_hass()
+
+    def on_attr_change(self) -> None:
+        """Handle attribute changes."""
+        now = utcnow()
+        cook_time_duration = self.oven._get_attribute(  # noqa: SLF001
```

Comment:
`_get_attribute` is an internal method and should not be used here. If `TimeSetCookTimeSet` is needed, then a method should be added on the library instead

Reply:
my suggestion for now is to just remove this sensor, so we can move forward with the PR.

Reply:
Removed for now